besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -937,12 +940,20 @@ private void addPluginMetrics(final BesuController<?> besuController) {
 
     final Object consensusState = besuController.getProtocolContext().getConsensusState();
 
-    if (consensusState != null && PoAContext.class.isAssignableFrom(consensusState.getClass())) {
-      final PoAMetricServiceImpl service =
-          new PoAMetricServiceImpl(
-              ((PoAContext) consensusState).getBlockInterface(),
-              besuController.getProtocolContext().getBlockchain());
-      besuPluginContext.addService(PoAMetricsService.class, service);
+    if (consensusState != null) {
+      if (IbftContext.class.isAssignableFrom(consensusState.getClass())) {
```

Comment:
I know you're following the existing pattern here, but why are we doing `isAssignableFrom` checks here?  I would have expected this to be pushed into the `BesuController` class to register these services as it has the correct context types and doesn't need any casting.

Reply:
Yeah - way nicer